gax-java/gax/src/main/java/com/google/api/gax/retrying/ExponentialRetryAlgorithm.java

```diff
@@ -67,10 +67,16 @@ public ExponentialRetryAlgorithm(RetrySettings globalSettings, ApiClock clock) {
    */
   @Override
   public TimedAttemptSettings createFirstAttempt() {
+    long totalTimeout = globalSettings.getTotalTimeout().toMillis();
+    Duration initialTimeout =
+        totalTimeout == 0
+            ? globalSettings.getInitialRpcTimeout()
```

Comment:
Done! Also created a issue https://github.com/googleapis/sdk-platform-java/issues/2643

Reply:
qq, do you know if there is a valid use case where the total timeout is non-zero (explicitly set by the user), but the the total timeout is set to be less than the initialRPCTimeout? The only thing that I can think of right now is that a user has both time and attempt bounds and they care more about the attempt bounds (maybe?). Even if that is the case, the initialRPCTimeout shouldn't be larger than the totalTimeout.

After some more thought, I'm slightly leaning towards having a check that ensures that the initialRPCTimeout is less than or equal to the total timeout if totalTimeout is set. It *should* eliminate the need for a check here. Something here: https://github.com/googleapis/sdk-platform-java/blob/e542af8a8548fa336e2d3cd15111a0672358c84a/gax-java/gax/src/main/java/com/google/api/gax/retrying/RetrySettings.java#L429-L452, but that *may* be a behavior breaking change.

CC: @blakeli0

Reply:
Yeah, we have a customer that only sets total timeout to be 400ms. The default initialRpcTimeout in bigtable client is 20 seconds. I think adding the check here would cause the breakage for the customer.

Reply:
Gotcha, I'm guessing they also either override the initialRpcTimeout or they're fine that call only runs a max of 400ms. I don't know the full context and there may be a use case where intialRpcTimeout > totalTimeout. Either way, I think you're right and we can't make that change. I think this solution is fine with me then.